src/core/server/ui_settings/types.ts

```diff
@@ -102,6 +102,8 @@ export interface UiSettingsParams {
   readonly?: boolean;
   /** defines a type of UI element {@link UiSettingsType} */
   type?: UiSettingsType;
+  /** field ui options */
+  uiOptions?: Record<string, any>;
```

Comment:
`uiOptions` sounds like anything used in UI, although reporting uses it for validation. There is the `validation` field for the same purpose added in https://github.com/elastic/kibana/pull/44678 although it's missed in TypeScript definition. Would you mind adding one?
We have the separate issue to introduce a proper validation for UI settings https://github.com/elastic/kibana/issues/46717
Thus, we should mark this validation field as deprecated.
```ts
type StringValidation = {
  regexString: string,
  message: string
};
type ImageValidation = {
  maxSize: ...
}
interface UiSettingsParams {
...
/*
* Allows defining a custom validation applicable to value change on the client.
* @deprecated 
*/
validation: StringValidation | ImageValidation;
}
```